rules/ruby/third_parties/appsignal.yml

```
@@ -0,0 +1,95 @@
+imports:
+  - ruby_shared_common_datatype
+patterns:
+  - pattern: |
+      Appsignal.$<METHOD>($<...>$<DATA_TYPE>$<...>)
+    filters:
+      - variable: DATA_TYPE
+        detection: ruby_shared_common_datatype
+        scope: result
+      - variable: METHOD
+        values:
+          - set_action
+          - add_breadcrumb
+          - add_exception
+          - set_error
+          - set_metadata
+          - set_namespace
+          - set_tags
+          - tag_job
+          - tag_request
+  - pattern: |
+      Appsignal.$<ERROR_METHOD>($<...>) do |$<TRANSACTION:identifier>|
+        $<TRANSACTION>.$<METHOD>($<...>$<DATA_TYPE>$<...>)
```

You probably want this so that if that triggers, it will point directly to the correct line and not the entire block. (see code change in test below)

```suggestion
        $<!>$<TRANSACTION>.$<METHOD>($<...>$<DATA_TYPE>$<...>)
```